### DISCUSSION TITLE
Bookmark which uses clipboard text as search term?

### DISCUSSION THREAD

**Discussion by letmeindude** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#89] [Q&A]
Is it possible to make a **Bookmark** which will use clipboard content as search term?

In Pale Moon browser I use XUL extension which allows special bookmarks like:

https://www.imdb.com/find/?q={clipTerms}

**Is this possible with SPC and Bookmarklet API?**

I know I need to use spc.clipboard, but I'm not sure how.

P.S. Amazing extension, thank you very much.

**Environment:**
OS [Windows 10]
Browser [Firefox 124.0.2]
Extension Version [8.1]

**Comment by YoshifumiFuyuno** [comment]
Hi.

If you search with the normal search engine without selecting a string, the search will be performed with the contents of the clipboard.
* Search by right-clicking without selecting anything
* Search from popup. There are several ways to do this
  * Search from Ctrl+Shift+Space without selecting anything ( https://support.mozilla.org/kb/manage-extension-shortcuts-firefox )
  * Search from toolbar button without selecting anything
  * Select a string and delete the string displayed in the popup before searching

If you use the Bookmarklet API, it will look like this:
```javascript
javascript:spc.clipboard.readText().then(c => spc.spc.openURL( {url: "https://www.imdb.com/find/?q=" + encodeURIComponent(c.trim())} ));
```

Best regards

**Reply by letmeindude** [reply]
Thank you, unfortunately nothing happens when I click on this **bookmark**:
`javascript:spc.clipboard.readText().then(c => spc.spc.openURL( {url: "https://www.imdb.com/find/?q=" + encodeURIComponent(c.trim())} ));`

It seems **regular bookmarks** can't do this on their own, I hope you will consider this feature.
